src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -627,6 +629,219 @@ class Polyhedron
+    poly.addVertex(oct[2]);
+    poly.addVertex(oct[3]);
+    poly.addVertex(oct[4]);
+    poly.addVertex(oct[5]);
+
+    poly.addNeighbors(0, {1, 5, 4, 2});
+    poly.addNeighbors(1, {0, 2, 3, 5});
+    poly.addNeighbors(2, {0, 4, 3, 1});
+    poly.addNeighbors(3, {1, 2, 4, 5});
+    poly.addNeighbors(4, {0, 5, 3, 2});
+    poly.addNeighbors(5, {0, 1, 3, 4});
+
+    // Reverses order of vertices 1,2 and 4,5 if volume is negative.
+    // Expanded swap operations for HIP workaround.
+    if(checkSign)
+    {
+      double vol = poly.volume();
+      if(vol < 0)
+      {
+        PointType p2({poly[4][0], poly[4][1], poly[4][2]});
+        poly[4][0] = poly[5][0];
+        poly[4][1] = poly[5][1];
+        poly[4][2] = poly[5][2];
+        poly[5][0] = p2[0];
+        poly[5][1] = p2[1];
+        poly[5][2] = p2[2];
+
+        PointType p1({poly[1][0], poly[1][1], poly[1][2]});
+        poly[1][0] = poly[2][0];
+        poly[1][1] = poly[2][1];
+        poly[1][2] = poly[2][2];
+        poly[2][0] = p1[0];
+        poly[2][1] = p1[1];
+        poly[2][2] = p1[2];
+      }
+    }
+
+    return poly;
+  }
+
+  /*!
+ * \brief Creates a Polyhedron from a given Tetrahedron's vertices.
+ *
+ * \param [in] tet The tetrahedron
+ * \param [in] checkSign If true (default is false), checks the volume of the
+ *             Polyhedron is positive. If volume is negative, order of some
+ *             vertices will be swapped.
+ *
+ * \return A Polyhedron with the Tetrahedron's vertices and added
+ *         vertex neighbors
+ *
+ * \note The Tetrahedron is assumed to have a specific vertex order:
+ * \verbatim
+ *
+ *              3                    +y
+ *             / \\                       +z
+ *            /   \ \                 ^  >
+ *           /     \  \               | /
+ *          /       \   \             |/
+ *         /         \    2           -----> +x
+ *        /           \  /
+ *       /_____________\/
+ *      0               1
+ *
+ * \endverbatim
+ *
+ *       The Polyhedron's vertex neighbors are created assuming this vertex
+ *       ordering.
+ *
+ * \note checkSign flag does not guarantee the Polyhedron's vertex order
+ *       will be valid. It is the responsiblity of the caller to pass
+ *       a Tetrahedron with a valid vertex order.
+ */
+  AXOM_HOST_DEVICE
+  static Polyhedron from_primitive(const Tetrahedron<T, NDIMS>& tet,
+                                   bool checkSign = false)
+  {
+    // Initialize our polyhedron to return
+    Polyhedron<T, NDIMS> poly;
+
+    poly.addVertex(tet[0]);
+    poly.addVertex(tet[1]);
+    poly.addVertex(tet[2]);
+    poly.addVertex(tet[3]);
+
+    poly.addNeighbors(0, {1, 3, 2});
+    poly.addNeighbors(1, {0, 2, 3});
+    poly.addNeighbors(2, {0, 3, 1});
+    poly.addNeighbors(3, {0, 1, 2});
+
+    // Reverses order of vertices 1 and 2 if volume is negative
+    if(checkSign)
+    {
+      if(poly.volume() < 0)
+      {
+        axom::utilities::swap<PointType>(poly[1], poly[2]);
```

Comment:
Do the neighbors not need to be swapped?

Reply:
Minor: For tetrahedra, would it be better to check `tet.signedVolume()` before setting up the polyhedron?